### PR TITLE
Use a separate control parameter to print out action dependencies

### LINKED ISSUE #17929: Add a separate parameter from Debug/show_actions for showing the action dependencies
## Reason
`Debug/show_actions` is a quit use parameter that can show the sequence of action setups. It also activates a print-out of action dependencies before showing the execution of actions, which is less useful to users.

## Design
We can add a separate control parameter for printing the action dependencies and let `show_actions` only show the execution of all the actions.

## Impact
Better control on the screen print-out thus making the user interface a little more friendly.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #17929.

### CODE REVIEW

**Comment by YaqiWang** [comment]
This is ready for review or merge. Thanks.

File: framework/include/actions/ActionWarehouse.h [hunk 1]
```diff
@@ -202,6 +202,8 @@ class ActionWarehouse : public ConsoleStreamInterface
    * various warehouse operations during the problem setup phase.
    * @param state Flag indicating whether to show action information.
    */
+  void showActionDependencies(bool state = true) { _show_action_dependencies = state; }
+
```
Comment (loganharbour) [inline framework/include/actions/ActionWarehouse.h:206]: Make doc strings appropriate, given that there is not one at the variable definition

File: test/tests/actions/debug_block/debug_print_actions_test.i [hunk 2]
```diff
@@ -44,6 +44,7 @@
 []
 
 [Debug]
+  show_action_dependencies = true
```
Comment (loganharbour) [inline test/tests/actions/debug_block/debug_print_actions_test.i:47]: Please change the test specifications; we should actually have one test individually for each of these methods and should not run them all at once

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: framework/include/actions/ActionWarehouse.h (hunk 1 above)
Comment (YaqiWang) [reply framework/include/actions/ActionWarehouse.h:206]: There is one docstring supposed for these three functions. I guess we need to add something like `///@{`.

File: test/tests/actions/debug_block/debug_print_actions_test.i (hunk 2 above)
Comment (YaqiWang) [reply test/tests/actions/debug_block/debug_print_actions_test.i:47]: The `tests` actually contains the two exact specifications we want.

**Comment by YaqiWang** [comment]
Docstrings are updated.

File: test/tests/actions/debug_block/debug_print_actions_test.i (hunk 2 above)
Comment (YaqiWang) [reply test/tests/actions/debug_block/debug_print_actions_test.i:47]: And the same input is run twice to test the two specifications.

**Comment by YaqiWang** [comment]
@loganharbour any further comments? That failure seems irrelevant.

**Review by loganharbour — APPROVED** [review]

**Pull request merged**